### ISSUE TITLE
Case-insensitive property access

### ISSUE THREAD

**Issue by Waldo000000** [datalust/seq-tickets#894]
## Problem
Consider these two events:

```
log.Information("{userId}", 1); // camelCase property name
log.Information("{UserId}", 1); // PascalCase property name
```

To create a filter that matches both of these events, this [seemingly](https://docs.datalust.co/docs/query-syntax) requires the following syntax:
```
userId = 1 or UserId = 1
```

There are two problems with this:
1. There are some scenarios where a mixture of such log events may occur unintentionally (e.g. when ingesting log events or creating filters from a mixture of client-side and server-side languages). As a result, users might accidentally miss log events when running queries which they assume are case-insensitive for property names
2. The syntax `userId = 1 or UserId = 1` is verbose/repetitive and more difficult to read, particularly when the filter includes many different property names

## Proposal
Two ideas:
1. Allow for the server behavior to be case-insensitive for property names, either by default or via a server setting. In this case, searching for simply `UserId = 1` would return both events from the example above.

2. Introduce a new syntax that explicitly allows users to indicate that they want the property names in the filter to be interpreted as case-insensitive. For example, `"UserId" = 1` may return both events from the example above, while `UserId = 1` would continue to only return the event that used PascalCase. (the double quotes here are just an example; other symbols or operators could be used in their place to the same effect)

**Comment by nblumhardt** [comment]
Thanks for writing this up, Roy 👍 . We've been doing some thinking about case-(in)sensitivity lately, and it's good to have this scenario to consider.

**Comment by nblumhardt** [comment]
I've added some ideas on how we could approach this in: https://github.com/nblumhardt/seq-tickets/blob/ci-modifier-rfc/rfcs/0000-ci-modifier.md#case-insensitive-property-access - would appreciate any feedback you might have on that direction :-) Cheers!

**Comment by nblumhardt** [comment]
Now in-scope for #919.

**Comment by nblumhardt** [comment]
This is implemented as:

```sql
ElementAt(@Properties, 'userid') ci
```

**Comment by nblumhardt** [comment]
Included in the just-published 6.0.3455-pre MSI and Docker container 👍
